scripts/build-doc.sh

```diff
@@ -0,0 +1,4 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p entr gnumake mandoc pandoc"
+
+make -f scripts/doc.Makefile
```

Comment:
We're using makefile in combination with nix here, which I think is OK; but within nix, we can possibly remove the need for a makefile altogether.

Reply:
See my above comment. I think the `Makefile` is cool to re-build just what's needed during development. Btw, `make` is an implicit dependency of any Nix environment (e.g. `nix-shell -p hello --run 'make -h'` will work) and I think it's ok to use `make` within the Nix derivation. But we can get rid of it if you think we can still have a nice developer experience (i.e. rebuild only what's needed).